tests/test_isPrimaryFlag.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+from lsst.utils import getPackageDir
+from lsst.log import Log
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask, CharacterizeImageConfig
+from lsst.pipe.tasks.calibrate import CalibrateTask, CalibrateConfig
+
+
+class IsPrimaryTestCase(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        # Load sample input from disk
+        expPath = os.path.join(getPackageDir("pipe_tasks"), "tests", "data", "v695833-e0-c000-a00.sci.fits")
+        self.exposure = afwImage.ExposureF(expPath)
+        # set log level so that warnings do not display
+        Log.getLogger("calibrate").setLevel(Log.ERROR)
+
+    def tearDown(self):
+        del self.exposure
+
+    def testIsPrimaryFlag(self):
+        """Tests detect_isPrimary column gets added when run.
+        """
+        charImConfig = CharacterizeImageConfig()
+        charImTask = CharacterizeImageTask(config=charImConfig)
+        charImResults = charImTask.run(self.exposure)
+        calibConfig = CalibrateConfig()
+        calibConfig.doAstrometry = False
+        calibConfig.doPhotoCal = False
+        calibTask = CalibrateTask(config=calibConfig)
+        calibResults = calibTask.run(charImResults.exposure)
+        outputCat = calibResults.outputCat
+        self.assertTrue("detect_isPrimary" in outputCat.schema.getNames())
+        # make sure all sky sources are flagged as not primary
+        self.assertTrue(sum((outputCat["detect_isPrimary"]) & (outputCat["sky_source"])) == 0)
+        # make sure all parent sources are flagged as not primary
+        self.assertTrue(sum((outputCat["detect_isPrimary"]) & (outputCat["deblend_nChild"] > 0)) == 0)
```

Comment:
Maybe this is efficient, but asserting the sum of two things being `&`-ed should be equal to zero is ... not a very intuitive check to me. Could you please break this down a bit, or explain why you did it this way?

Reply:
These are bitwise ANDs for two boolean lists, i.e. (for line 59 above), we're checking that zero objects are both `detect_isPrimary == True` and `sky_source == True`. If any object has been labelled as a sky source, then it should not also be labelled as primary. Does that make sense? Happy to add some extra comments to the code if you think it might be clearer?

Reply:
Thanks, the sum is what threw me off.

It might read a little easier as a few assertEqual statements instead? e.g., `self.assertEqual(sum(outputCat["detect_isPrimary"]) & (outputCat["sky_source"]), 0)` ?

Reply:
That looks great, thanks for the suggestion - both asserts changed.